Formula/f/foldseek.rb

```diff
@@ -0,0 +1,50 @@
+class Foldseek < Formula
+  desc "Search your protein structures in seconds"
+  homepage "https://github.com/steineggerlab/foldseek"
+  url "https://github.com/steineggerlab/foldseek/archive/8-ef4e960.tar.gz"
+  version "8-ef4e960"
+  sha256 "c74d02c4924d20275cc567783b56fff10e76ed67f3d642f53c283f67c4180a1e"
+  license "GPL-3.0-or-later"
+  head "https://github.com/steineggerlab/foldseek.git", branch: "master"
+
+  depends_on "cmake" => [:build, :test]
+  depends_on "rust" => :build
+
+  uses_from_macos "bzip2"
+  uses_from_macos "zlib"
+
+  on_macos do
+    depends_on "libomp"
+  end
+
+  resource "testdata" do
+    url "https://github.com/steineggerlab/foldseek-regression/archive/361fa799ed94cba67accd38afa380f3bc0703b0e.tar.gz"
+    sha256 "c5c1afc0e4bdcd411ec139ee80fff1c3b00ec69a732f5bb549f41395f3505c9b"
+  end
+
+  def install
+    # fix "DSO missing from command line" error
+    ENV.append "LDFLAGS", "-Wl,--copy-dt-needed-entries" if OS.linux?
+    args = *std_cmake_args
+    if OS.mac?
+      libomp = Formula["libomp"]
+      args << "-DOpenMP_C_FLAGS=-Xpreprocessor -fopenmp -I#{libomp.opt_include}"
+      args << "-DOpenMP_C_LIB_NAMES=omp"
+      args << "-DOpenMP_CXX_FLAGS=-Xpreprocessor -fopenmp -I#{libomp.opt_include}"
+      args << "-DOpenMP_CXX_LIB_NAMES=omp"
+      args << "-DOpenMP_omp_LIBRARY=#{libomp.opt_lib}/libomp.a"
+    end
+
+    system "cmake", "-S", ".", "-B", "build", *args
+    system "cmake", "--build", "build"
+    system "cmake", "--install", "build"
+
+    pkgshare.install "example"
+  end
+
+  test do
+    resource("testdata").stage do
+      system "./run_regression.sh", "#{bin}/foldseek", "tmp"
```

Comment:
Is this something users will usually run?

We need a test that exercises the some of the functionality of the app.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
I suggested this test in reference to a formula https://github.com/Homebrew/homebrew-core/blob/master/Formula/m/mmseqs2.rb, which is created by the same developers. Is it appropriate? If not, I will consider another way.

Reply:
Well, does it exercise some of the functionality in the way users do? Because that is the main criterium for a good Homebrew test block.

Reply:
Thank you for your detailed explanation. I modified the test block to use the simple test according to the official example. https://github.com/steineggerlab/foldseek/tree/master#search